pilot/pkg/networking/plugin/authz/rbac.go

```diff
@@ -599,20 +604,6 @@ func permissionForKeyValues(key string, values []string) *policyproto.Permission
 
 func principalForKeyValue(key, value string) *policyproto.Principal {
 	switch {
-	case key == userHeader:
-		if value == "*" {
-			// Generate an any rule to grant access permission to anyone if the value is "*". This is the
-			// only wildcard character we could support for principal.user.
-			return &policyproto.Principal{
-				Identifier: &policyproto.Principal_Any{
-					Any: true,
-				},
-			}
-		}
```

Comment:
We should still special handle the case that `user: "*"`. It means that all users (authenticated or unauthenticated) are allowed. We need to map it to "principals: - any: true". 

If `user` equals to any value that is NOT "*", `user` is equivalent to `source.principal`. 

`source.principal: "*"` means that all authenticated users are allowed. This is because the value of `source.principal` must exist (set by the authentication filter), but the value can be anything. 

`user: "*"` means that all users are allowed, regardless it is authenticated or not. 

In the future, we may introduce specific enums for these two cases to make it clear.

Reply:
Done. I think we should update the document for this difference.